nixos/modules/services/video/mirakurun.nix

```diff
@@ -0,0 +1,155 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.mirakurun;
+  mirakurun = pkgs.mirakurun;
+  username = config.users.users.mirakurun.name;
+  groupname = config.users.users.mirakurun.group;
+  settingsFmt = pkgs.formats.yaml {};
+
+in
+  {
+    options = {
+      services.mirakurun = {
+        enable = mkEnableOption mirakurun.meta.description;
+
+        logLevel = mkOption {
+          type = types.ints.between (-1) 3;
+          default = 2;
+          description = "Mirakurun log level.";
+        };
+
+        domainSocket = mkOption {
+          type = with types; nullOr path;
+          default = "/var/run/mirakurun/mirakurun.sock";
+          description = "Path to domain socket used to communicate with Mirakurun.";
+        };
+
+        port = mkOption {
+          type = with types; nullOr port;
+          default = 40772;
+          description = "Port to listen on.";
+        };
+
+        serverSettings = mkOption {
+          type = settingsFmt.type;
+          default = {};
+          example = literalExample ''
+            {
+              highWaterMark = 25165824;
+              overflowTimeLimit = 30000;
+            };
+          '';
+          description = "Extra options which are added to server.yml.";
+        };
+
+        tunerSettings = mkOption {
+          type = with types; nullOr (listOf attrs);
```

Comment:
`settingsFmt` can be used here and for `channelSettings` as well

Reply:
I've now changed it use `settingsFmt` for file generation. Would it be acceptable to keep the types as is? It'd provide a somewhat stronger typing guarantee.

Reply:
I think we should still use the generic yaml type for this, because one motivation for such `settings` options is that the module is still usable even if upstream adds new program settings. If we restrict this to only a subset of yaml, people can't use it if upstream ever has a setting that isn't in that subset.

Reply:
Maybe `listOf settingsFmt.type` would be okay though.

Reply:
It's now `settingsFmt.type`.